Applications/WebPortal/WebApp/library/source/pages/account_directory_page/account_directory_page/account_directory_controller.tsx

```diff
@@ -71,22 +73,32 @@ export class AccountDirectoryController extends
     this.props.model.load().then(
       () => {
         this.setState({
-          isLoaded: true
+          isLoaded: true,
+          sortedGroups: this.props.model.groups.sort(
+            AccountDirectoryController.groupComparator)
         });
       });
   }
 
-  public componentDidUpdate(): void {
+  public componentDidUpdate(prevProps: Properties): void {
     if(this.state.redirect) {
       this.setState({redirect: null});
     }
+    if(prevProps.model.groups.toString() !== 
```

Comment:
When is this needed? It's a very expensive operation to perform this, so I'm hoping there is a way to eliminate it.

Reply:
This would only be needed if a different user (or even the same one) added groups.

Otherwise I can just do a sort inside of onCreateGroup.